deeplake/core/vectorstore/deeplake_vectorstore.py

```diff
@@ -279,6 +309,12 @@ def add(
         assert id_ is not None
         utils.check_length_of_each_tensor(processed_tensors)
 
+        # In Case prefilled dataset which is already having index defined
+        # regenerate the index post ingestion.
+        index_regeneration = False
+        if len(self.dataset) > 0 and index.check_vdb_indexes(self.dataset):
+            index_regeneration = True
+
         dataset_utils.extend_or_ingest_dataset(
```

Comment:
@sounakr FYI Commits are created internally here, so I don't think removing  `self.dataset.commit(allow_empty=True)` is sufficient for fully removing version control.

cc @nvoxland